From 8a4783a2938b508dacf411742e7c0be41cec0190 Mon Sep 17 00:00:00 2001
From: Nicolas Goueslain <nicolas.goueslain@nxp.com>
Date: Wed, 23 Jul 2025 11:57:31 +0000
Subject: [PATCH] meson.build: Fix include path for numpy [YOCIMX-8735]

On walnascar, a numpy include path change causes a build break:

Check usable header "numpy/arrayobject.h" with dependency python3-embed:
NO
../git/meson.build:574: WARNING: Found python3, but failed to find
numpy.
../git/meson.build:575: WARNING: Disable nnstreamer-python3.
../git/meson.build:581:4: ERROR: Problem encountered: Cannot find
python3 with numpy support
ERROR: meson failed

Upstream-Status: Pending
Signed-off-by: Neena Busireddy <neenareddy.busireddy@nxp.com>
Signed-off-by: Nicolas Goueslain <nicolas.goueslain@nxp.com>
---
 meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index f5015b1c..e3a8f197 100644
--- a/meson.build
+++ b/meson.build
@@ -458,10 +458,10 @@ if not get_option('python3-support').disabled()
 
     python3_inc_args = []
     python3_inc_args += run_command(pg_pkgconfig, ['python3', '--cflags'], check : true).stdout().strip().split()
-    python3_inc_args += run_command('python3', ['-c', 'import site\nfor i in site.getsitepackages(): print("-I" + i + "/numpy/core/include")'], check : true).stdout().strip().split()
+    python3_inc_args += run_command('python3', ['-c', 'import site\nfor i in site.getsitepackages(): print("-I" + i + "/numpy/_core/include")'], check : true).stdout().strip().split()
     r = run_command('python3', ['-m', 'site', '--user-site'], check : false)
     if r.returncode() == 0
-      python3_inc_args += '-I' + r.stdout().strip() + '/numpy/core/include'
+      python3_inc_args += '-I' + r.stdout().strip() + '/numpy/_core/include'
     endif
     python3_inc_valid_args = []
 
-- 
2.34.1

